### ISSUE TITLE
Calico Node requires IPv4.

### ISSUE THREAD

**Issue by valentin2105** [projectcalico/calico#1204]
Hi,

I'm currently deploying an IPv6 Only Kubernetes cluster, 

I'm facing a small problem with `Calico Node` because it requires inevitably an IPv4 Address to identify the node. ( `-e IP= ` )

## Possible Solution

Make possible to disable totally IPv4 connexions with a Calico node argument or environment variables ?

## Your Environment
* Calico version `calico/node:latest` 
* Orchestrator version (e.g. kubernetes, mesos, rkt): `v1.8.1`
* Operating System and version: Debian 9.1

**Comment by caseydavenport** [comment]
@valentin2105 could you provide the startup logs from `calico/node` in your environment? Would be interested to see exactly where startup fails.

**Comment by valentin2105** [comment]
Sure,

When I try to launch it without any IPv4 : 
```
Status: Downloaded newer image for calico/node:latest
2017-10-17 21:37:53.369964 I | warning: ignoring ServerName for user-provided CA for backwards compatibility is deprecated
Skipping datastore connection test
WARNING: Unable to auto-detect an IPv4 address: no valid IPv4 addresses found on the host interfaces
ERROR: Couldn't autodetect a management IPv4 address:
  -  provide an IPv4 address by configuring one in the node resource, or
  -  provide an IPv4 address using the IP environment, or
  -  if auto-detecting, use a different autodetection method.
```

**Comment by caseydavenport** [comment]
@valentin2105 ok cool, thanks.

Looks like the relevant code is here: https://github.com/projectcalico/calico/blob/master/calico_node/startup/startup.go#L265-L291

We probably want to support an `IP=none` or  maybe an `IP_AUTODETECTION_METHOD=none` option so that this can be disabled.

**Comment by caseydavenport** [comment]
Alternatively, we just mimic what's done for IPv6 and require `IP=autodetect` for ipv4 as well. This is a breaking change though so we need to be careful about that. I think it's safest to assume IPv4 and only disable it when explicitly told so.

**Comment by valentin2105** [comment]
Yeah, I think disabling IPv4 should be a manual process, `IP=none` seem great. 
So it assume Calico node will try to stop peering with other nodes in IPv4 too ?! 

Thanks !

**Comment by valentin2105** [comment]
@caseydavenport  Any advancement on this ? 'cause the 2.6.3 is released. 
I can try to make a PR if it doesn't.

**Comment by caseydavenport** [comment]
@valentin2105 a PR would be lovely - we're pretty stretched at the moment and haven't had a chance to get to this ourselves, but I'd really like to see this happen.

**Comment by tmjd** [comment]
Just wanted to add some new info here from another issue.
https://github.com/projectcalico/calico/pull/1736#issuecomment-371325385

TL;DR
The problem still exists with Calico v3.0.3 and K8s v1.9.3.

**Comment by valentin2105** [comment]
It seem the latest issue for IPv6 Calico Features (https://github.com/projectcalico/calico/issues/595).

I think to make it happend we need  to allow the Node's IPv4 to be empty : 
```
  spec:
    bgp:
      ipv4Address: none
      ipv6Address: ff9b::/128
```
Is that possible in the Calico structure ?

**Comment by tmjd** [comment]
I'm actually working on this currently.  I've tested disabling the detection of an IPv4 address and currently the problem I'm hitting is that the ipv4 address is used as the router id in bird(6) configuration, unfortunately it is not as simple as just using the ipv6 address instead.

**Comment by valentin2105** [comment]
Yeah it's true, If i'm not wrong, we can't use an IPv6 as routerID. 
Maybe this like will help https://networkengineering.stackexchange.com/questions/510/how-to-choose-a-bgp-router-id-when-using-ipv6-only

**Comment by tmjd** [comment]
Yeah I've seen that (thanks for it though), what I've been going from based on that is that we just need a unique 32bit value to use for the router ID in the cluster.
Next step is to figure out what we can use for that or if a new field has to be added to the datastore for that purpose.

**Comment by caseydavenport** [comment]
@tmjd is this fixed now?

**Comment by tmjd** [comment]
I believe IPv4 is not required now (though I didn't have a pure IPv6 environment to test in).  See https://docs.projectcalico.org/master/usage/ipv6#using-only-ipv6 for the necessary configuration.

**Comment by valentin2105** [comment]
@tmjd  Is great, thank you !!

I think it will be nice if we can generate automaticly RouterID without any IPv4, because while deploying calico in DaemonSet, we can't change the ENV variable of router ID for each workers. 

Any though about that ?

**Comment by tmjd** [comment]
Yeah it would be great to have that automatically generated.
I think we would have to generate the RouterID and then store it in the datastore to make sure we don't have collisions, unless there is some mechanism for discovering a conflict in the cluster.
It was decided to do this first step and that it would be an improvement to design and implement the automated version.  We're open to input on this and happy to accept PRs if anyone is motivated to do this work.

**Comment by caseydavenport** [comment]
@tmjd I think this is no longer the case?
